File: IS/front/src/ts/data/OrganizationType.ts
enum OrganizationType {
    COMMERCIAL = 'COMMERCIAL',
    GOVERNMENT = 'GOVERNMENT',
    TRUST = 'TRUST',
    PRIVATE_LIMITED_COMPANY = 'PRIVATE_LIMITED_COMPANY',
    OPEN_JOINT_STOCK_COMPANY = 'OPEN_JOINT_STOCK_COMPANY'
}

export const organizationTypeToString = new Map<OrganizationType, string>()
organizationTypeToString.set(OrganizationType.COMMERCIAL, "Коммерческая")
organizationTypeToString.set(OrganizationType.GOVERNMENT, "Правительственная")
organizationTypeToString.set(OrganizationType.TRUST, "Траст")
organizationTypeToString.set(OrganizationType.PRIVATE_LIMITED_COMPANY, "Частная компания")
organizationTypeToString.set(OrganizationType.OPEN_JOINT_STOCK_COMPANY, "Открытое акционерное общество")

export const stringToOrganizationType = (orgType: string) => {
    switch (orgType) {
        case 'GOVERNMENT':
            return OrganizationType.GOVERNMENT
        case 'TRUST':
            return OrganizationType.TRUST
        case 'PRIVATE_LIMITED_COMPANY':
            return OrganizationType.PRIVATE_LIMITED_COMPANY
        case 'OPEN_JOINT_STOCK_COMPANY':
            return OrganizationType.OPEN_JOINT_STOCK_COMPANY
        default:
            return OrganizationType.COMMERCIAL
    }
}


export default OrganizationType

File: back/src/main/java/com/example/back/model/dto/OrganizationType.java
package com.example.back.model.dto;

public enum OrganizationType {
    COMMERCIAL,
    GOVERNMENT,
    TRUST,
    PRIVATE_LIMITED_COMPANY,
    OPEN_JOINT_STOCK_COMPANY;

    public static OrganizationType stringToOrganizationType(String s) {
        return switch (s.toLowerCase()) {
            case "commercial" -> COMMERCIAL;
            case "government" -> GOVERNMENT;
            case "trust" -> TRUST;
            case "private_limited_company" -> PRIVATE_LIMITED_COMPANY;
            case "open_joint_stock_company" -> OPEN_JOINT_STOCK_COMPANY;
            default -> null;
        };
    }
}